File: compiler/testData/loadJava/compiledJavaCompareWithKotlin/kotlinSignature/propagation/return/TwoSuperclassesReturnJavaSubtype.java
package test;

import org.jetbrains.annotations.NotNull;
import jet.runtime.typeinfo.KotlinSignature;

public interface TwoSuperclassesReturnJavaSubtype {

    public interface Super1 {
        public CharSequence foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Super2 {
        @NotNull
        public CharSequence foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Sub extends Super1, Super2 {
        public String foo();
    }
}

File: compiler/testData/loadJava/compiledJavaCompareWithKotlin/kotlinSignature/propagation/return/TwoSuperclassesReturnJavaSubtype.kt
package test

public trait TwoSuperclassesReturnJavaSubtype {

    public trait Super1 {
        public fun foo(): CharSequence?

        public fun dummy() // to avoid loading as SAM interface
    }

    public trait Super2 {
        public fun foo(): CharSequence

        public fun dummy() // to avoid loading as SAM interface
    }

    public trait Sub: Super1, Super2 {
        override fun foo(): String
    }
}
